pysat/_params.py

```diff
@@ -262,25 +264,20 @@ def _set_data_dirs(self, path=None, store=True):
         # Account for the presence of $HOME or similar
         paths = [os.path.expandvars(path) for path in paths]
 
-        # Ensure paths have some consistency despite user input
-        paths = [os.path.normpath(path) for path in paths]
+        # Make sure paths don't end with path separator for consistency
+        paths = [path if path[-1] != os.path.sep else path[:-1]
+                 for path in paths]
 
-        # Ensure all paths are valid
-        paths_check = [os.path.isdir(path) for path in paths]
+        # Ensure all paths are valid, create if not
+        [check_and_make_path(path) for path in paths]
```

Comment:
How does this line work?

Reply:
I'm using the list comprehension to simply iterate over the path elements and call the `check_and_make_path` function. It performs the check and makes the path as necessary.

Reply:
In that case, I'd recommend doing this as a for loop.  This will make it clear that you're running a series of tests and from previous time testing a simple loop and simple list compression take about the same time to run.